releaseNotes/Microsoft.Featuremanagement.AspNetCore.md

```diff
@@ -0,0 +1,20 @@
+# Microsoft.FeatureManagement.AspNetCore
+
+[Source code][source_code] | [Package (NuGet)][package] | [Samples][samples] | [Product documentation][docs]
+
+## 4.0.0-preview2 - March 7, 2024
+
+### Enhancements
+
+* Added `TargetingHttpContextMiddleware`, which ensures TargetingContext is available for syncronous functions, like `TargetingTelemetryInitializer` in the `Microsoft.FeatureManagement.Telemetry.ApplicationInsights.AspNetCore` package. ([#409](https://github.com/microsoft/FeatureManagement-Dotnet/issues/409))
```

Comment:
I found this one was hard to describe (maybe we should rethink the design). Yours is probably more technically correct, but also requires a lot of contexts to understand. How do you like this?

Introduced a new ASP.NET Core middleware called `TargetingHttpContextMiddleware`. It captures the targeting information for each incoming request. Use it with the `TargetingTelemetryInitializer` to add targeting information to telemetry data sent to Application Insights.

Reply:
I'm still not sure what the best wording is here. 

Your version and mine both have the same problem to me- it feels like we shouldn't mention `TargetingTelemetryInitializer` or App Insights in this package- since this the main "AspNetCore" package and customers using it may have no idea or care what `TargetingTelemetryInitializer` is.

I tried to remedy that by calling out what package the initializer is from, but looking back I don't think it's helpful. I think I prefer just saying exactly what it does:

Introduced a new ASP.NET Core middleware called `TargetingHttpContextMiddleware`. It makes `TargetingContext` available to synchronous functions without blocking.

Reply:
> It ensures `TargetingContext` is available to synchronous functions without blocking.

If we could get the TargetingContext in TargetingTelemetryInitializer, will we still introduce the middleware? My answer is no. So our goal is not to introduce a generic solution for "synchronous functions without blocking". We just need a solution for TargetingTelemetryInitializer. The "synchronous functions without blocking" is more of an implementation detail. This is why I suggested that we called out that the TargetingHttpContextMiddleware and TargetingTelemetryInitializer must work together and didn't make it sound like a generic solution in the release notes.

Reply:
I see what you mean. 

I still prefer trying to explain the feature rather than allowing release notes to be cross-package. What if we said even more specifically:

Introduced a new ASP.NET Core middleware called `TargetingHttpContextMiddleware`. It makes `TargetingContext` available from `HttpContext` on each request.

Reply:
Sure, we can go with this.

I'm thinking we probably should move TargetingHttpContextMiddleware to the same package as TargetingTelemetryInitializer. The TargetingHttpContextMiddleware is not intended to be used without the TargetingTelemetryInitializer.

Reply:
Sounds good. Created an issue for that idea: https://github.com/microsoft/FeatureManagement-Dotnet/issues/424